Firebase/ios/BonfireTests/PersistedChannelServiceTests.swift

```diff
@@ -0,0 +1,253 @@
+import XCTest
+import RxSwift
+import RxTests
+@testable import Bonfire
+
+
+class Remembrancer {
```

Comment:
We should probably move this stuff out into its own file. Also some context on what the hell this does.

Basically it's away to pretend-mock and keep track of method calls so we can check the call stack at the end of the test. It's far from perfect, but I think there's something in the idea. Or let's figure out how to write code where Mocks aren't needed 😆